tmt/result.py

```diff
@@ -109,26 +120,20 @@ def __init__(
                 self.ids[key] = test.node.get(key)
             interpret = ResultInterpret(test.result) if test.result else ResultInterpret.RESPECT
         else:
-            try:
-                self.ids = data['ids']
-            except KeyError:
-                self.ids = {}
+            self.ids = data.ids or {}
             interpret = ResultInterpret.RESPECT
 
         # Check for valid results
-        try:
-            self.result = ResultOutcome(data['result'])
-        except KeyError:
-            raise tmt.utils.SpecificationError("Missing test result.")
-        except ValueError:
-            raise tmt.utils.SpecificationError(
-                f"Invalid result '{data['result']}'.")
+        # try:
+        self.result = ResultInterpret(data.result)
```

Comment:
* are you sure about `ResultInterpret`? The original code suggests `result` should be `ResultOutcome`.
* `data.result` should already be of the right type, `ResultOutcome`, therefore simple `self.result = data.result` should take place, no need for converting `data.result` to an enum.